src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetProjectManagerService.cs

```diff
@@ -514,16 +509,40 @@ public async ValueTask<IReadOnlyCollection<IProjectContextInfo>> GetProjectsWith
             IEnumerable<Task<IProjectContextInfo>> tasks = affectedProjects
                 .Select(affectedProject => ProjectContextInfo.CreateAsync(affectedProject, cancellationToken).AsTask());
 
-            await Task.WhenAll(tasks);
-
-            return tasks.Select(task => task.Result).ToArray();
+            return await Task.WhenAll(tasks);
         }
 
         private static string CreateProjectActionId()
         {
             return Guid.NewGuid().ToString("N", CultureInfo.InvariantCulture);
         }
 
+        private async Task<IReadOnlyList<NuGetProject>> GetProjectsAsync(
+            IReadOnlyCollection<string> projectIds,
+            CancellationToken cancellationToken)
+        {
+            IVsSolutionManager? solutionManager = await _sharedState.SolutionManager.GetValueAsync(cancellationToken);
+            Assumes.NotNull(solutionManager);
+
+            Dictionary<string, NuGetProject>? projects = (await solutionManager.GetNuGetProjectsAsync())
+                .ToDictionary(project => project.GetMetadata<string>(NuGetProjectMetadataKeys.ProjectId), _ => _);
+            var matchingProjects = new List<NuGetProject>(capacity: projectIds.Count);
+
+            foreach (string projectId in projectIds)
+            {
+                if (projects.TryGetValue(projectId, out NuGetProject project))
+                {
+                    matchingProjects.Add(project);
+                }
+                else
+                {
+                    throw new ArgumentException("Project with ID {} not found.", nameof(projectIds));
```

Comment:
did you mean {0}

Reply:
Forgot to add a resource string.  Will fix.